logic/resolvable/Concludable.java

```diff
@@ -265,6 +265,17 @@ else if (a.getType().isDouble())
         return predicateFn;
     }
 
+    private static FunctionalIterator<AlphaEquivalence> alphaEqualValueConstraints(Set<ValueConstraint<?>> set1,
+                                                                                   Set<ValueConstraint<?>> set2) {
+        if (set1.size() != set2.size()) return Iterators.empty();
+        else {
+            for (ValueConstraint<?> s1 : set1) {
+                if (!Iterators.iterate(set2).flatMap(s1::alphaEquals).first().isPresent()) return Iterators.empty();
```

Comment:
fyi `iterate()` and `single()` are already statically imported.

Is it possible for two different value constraints to be alpha-equivalent to the same target? Eg. can many map to one? This line seems like there could be a hole where we don't exclude the constraints from `set2` that have already been mapped to when we check the next element in `set1` - is that a concern?

Reply:
You're right, and I was aware of this but failed to actually account for it. The hole is fixed now.